accord-core/src/main/java/accord/coordinate/Propose.java

```diff
@@ -102,32 +108,37 @@ public void onSuccess(Id from, AcceptReply reply)
         if (isDone)
             return;
 
-        switch (reply.outcome())
+        switch (reply.outcome)
         {
             default: throw new AssertionError("Unhandled AcceptOutcome: " + reply.outcome());
-            case Truncated:
-                isDone = true;
-                callback.accept(null, new Truncated(txnId, route.homeKey()));
-                break;
-
             case RejectedBallot:
                 isDone = true;
                 callback.accept(null, new Preempted(txnId, route.homeKey()));
                 break;
 
+            case Truncated:
             case Redundant:
-                if (reply.supersededBy != null || ballot.equals(Ballot.ZERO))
+                if (sendTo == route || !isSufficientPartialReply(reply, from))
                 {
-                    isDone = true;
-                    callback.accept(null, new Preempted(txnId, route.homeKey()));
+                    Throwable failNow = reply.outcome == AcceptOutcome.Truncated
```

Comment:
could we unwrap this ternary?

```
            if (reply.outcome == AcceptOutcome.Truncated)
                failNow = new Truncated(txnId, route.homeKey());
            else if (reply.supersededBy != null || ballot.equals(Ballot.ZERO))
                failNow = new Preempted(txnId, route.homeKey());
            else if (reply.committedExecuteAt != null)
                failNow = new Redundant(txnId, route.homeKey(), reply.committedExecuteAt);
            else
                failNow = null;
```

(please feel free to ignore if i am the only person who struggles parsing long ternaries)

Reply:
Sure. I think you have more of a problem with ternaries than me, but I am not adequately enamoured with them to prefer them strongly.